rankers/AggregateRanker/__init__.py

```diff
@@ -0,0 +1,32 @@
+from jina.executors.rankers import Chunk2DocRanker
+import numpy as np
+
+class AggregateRanker(Chunk2DocRanker):
+    """
+    :class:`AggregateRanker` calculates the score of the matched doc form the matched chunks. For each matched doc, the
+    score is aggregated from all the matched chunks belonging to that doc.
+    """
+
+    AGGREGATE_FUNCTIONS = ['min', 'max', 'mean', 'median', 'sum', 'prod']
+
+    def __init__(self, aggregate_function: str, is_reversed_score, *args, **kwargs):
+        """
+        :param: aggregate_function: defines the used aggregate function and can be one of:
+        [min, max, mean, median, sum, prod]
+        :param: is_reverse_score: True if a large score of the matched chunks mean a lower score of the matched doc,
+        othewise: False
+
+        """
+        super().__init__(*args, **kwargs)
+        self.is_reversed_score = is_reversed_score
+        if aggregate_function in self.AGGREGATE_FUNCTIONS:
+            self.np_aggregate_function = getattr(np, aggregate_function)
+        else:
+            raise ValueError(f'The aggregate function "{aggregate_function}" is not one of the values: "{self.AGGREGATE_FUNCTIONS}".')
```

Comment:
We could even be bold and do the following:
```
if hasattr(np, aggregate_function):
    if aggregate_function not in self.AGGREGATE_FUNCTIONS:
        self.logger.warning(f'The following aggregation function is only available for experimental use in jina and not fully tested: {aggregate_function}'
    self.np_aggregate_function = getattr(np, aggregate_functoin)
else:
   raise ...
```

Reply:
If we do this one could use var, std and let's say np.count_nonzero.
On the other hand... maybe I'm a bit paranoid, but to me this would feel like introducing a potenzial security vulnerability. If at the end the aggregate_function string comes from outside, one could call any np function. I don't have a concrete example where it is actually harmful, but if we just allow certain functions, we don't even have to think about that case.